activesupport/lib/active_support/callbacks.rb

```diff
@@ -693,14 +693,18 @@ def skip_callback(name, *filter_list, &block)
 
         __update_callbacks(name) do |target, chain|
           filters.each do |filter|
-            filter = chain.find {|c| c.matches?(type, filter) }
+            callback = chain.find {|c| c.matches?(type, filter) }
 
-            if filter && options.any?
-              new_filter = filter.merge_conditional_options(chain, if_option: options[:if], unless_option: options[:unless])
-              chain.insert(chain.index(filter), new_filter)
+            if !callback
+              warn "#{type.to_s.capitalize} #{name} callback #{filter.inspect} has not been defined"
             end
 
-            chain.delete(filter)
+            if callback && options.any?
+              new_callback = callback.merge_conditional_options(chain, if_option: options[:if], unless_option: options[:unless])
+              chain.insert(chain.index(callback), new_callback)
+            end
+
+            chain.delete(callback)
```

Comment:
Should this be in the `if callback` clause?

I think it'll be clearer if code looks like
```ruby
if callback
  if options.any?
    ... 
  end

  ...
else
  warn(...)
end